src/components/common/TokenSelect.tsx

```diff
@@ -120,22 +121,23 @@ export function TokenSelect({
 
   const [amountInCurrency, setAmountInCurrency] = useState<string>();
 
-  const decimals = selectedToken?.decimals || 18;
+  const decimals =
+    selectedToken && 'decimals' in selectedToken ? selectedToken.decimals : 18;
```

Comment:
In this particular component `selectedToken` can be null as well in which case we already defaulted to 18. 
Agreed, we probably should exclude the NFTs out of the type unions everywhere they shouldn't or can't get but opted to do it at another time since the PR is big enough already.